src/lib/Core.jsx

```diff
@@ -150,30 +173,61 @@ const Core = function ({
     if (filteredValue !== 'all') {
       if (filteredValue === 'correct') {
-        filteredQuestions = questions.filter((question, index) => correct.indexOf(index) !== -1);
-        filteredUserInput = userInput.filter((input, index) => correct.indexOf(index) !== -1);
+        filteredQuestions = questions.filter(
+          (question, index) => correct.indexOf(index) !== -1,
+        );
+        filteredUserInput = userInput.filter(
+          (input, index) => correct.indexOf(index) !== -1,
+        );
       } else {
-        filteredQuestions = questions.filter((question, index) => incorrect.indexOf(index) !== -1);
-        filteredUserInput = userInput.filter((input, index) => incorrect.indexOf(index) !== -1);
+        filteredQuestions = questions.filter(
+          (question, index) => incorrect.indexOf(index) !== -1,
+        );
+        filteredUserInput = userInput.filter(
+          (input, index) => incorrect.indexOf(index) !== -1,
+        );
       }
     }
 
     return (filteredQuestions || questions).map((question, index) => {
-      const userInputIndex = filteredUserInput ? filteredUserInput[index] : userInput[index];
+      const userInputIndex = filteredUserInput
+        ? filteredUserInput[index]
+        : userInput[index];
 
       // Default single to avoid code breaking due to automatic version upgrade
       const answerSelectionType = question.answerSelectionType || 'single';
 
       return (
-        <div className="result-answer-wrapper" key={index + 1}>
-          <h3 dangerouslySetInnerHTML={rawMarkup(`Q${question.questionIndex}: ${question.question}`)} />
-          {question.questionPic && <img src={question.questionPic} alt="image" />}
-          {renderTags(answerSelectionType, question.correctAnswer.length, question.segment)}
-          <div className="result-answer">
-            {renderAnswerInResult(question, userInputIndex)}
+        <>
+          <div className="result-answer-wrapper" key={index + 1}>
+            <h3
+              dangerouslySetInnerHTML={rawMarkup(
+                `Q${question.questionIndex}: ${question.question}`,
+              )}
+            />
+            {question.questionPic && (
+              <img src={question.questionPic} alt="image" />
+            )}
+            {renderTags(
+              answerSelectionType,
+              question.correctAnswer.length,
+              question.segment,
+            )}
+            <div className="result-answer">
+              {renderAnswerInResult(question, userInputIndex)}
+            </div>
+            <Explanation question={question} isResultPage />
           </div>
-          <Explanation question={question} isResultPage />
-        </div>
+          <div>
+            {index === (questions.length-1) ? (
+              <button onClick={() => location.reload()} className="red-button">
+                {appLocale.restartButton}
+              </button>
+            ) : (
+              ''
+            )}
+          </div>
```

Comment:
you don't need to put it here. you can put it after `{renderQuizResultQuestions()}`